src/Service/OrderStatusService.php

```diff
@@ -122,6 +123,10 @@ public function setOrderStatus($order, $statusId, $useExistingPayment = null, $t
             }
         }
 
+        if($status === Config::STATUS_PAID_ON_BACKORDER) {
+            $status = PaymentStatus::STATUS_PAID;
+        }
+
```

Comment:
I would like always even the smallest logic portion to be portable. This can be achieved by creating utility

`PaymentStatusUtility::getAsPossiblePaidStatus($status)` inside you can wrap same logic which would return status as paid and not in another case.